src/server/pkg/storage/fileset/api.go

```diff
@@ -24,7 +21,7 @@ var _ File = &FileReader{}
 // FileSet represents a set of files.
 type FileSet interface {
 	// Iterate iterates over the files in the file set.
-	Iterate(ctx context.Context, cb func(File) error) error
+	Iterate(ctx context.Context, cb func(File) error, deletive ...bool) error
```

Comment:
I don't like how there is a parameter, which I have 50/50 shot of getting wrong, that is also implicit.

Reply:
You could just as easily use the wrong iterate function if we exposed it as `IterateDeletes`. This approach allows for less boilerplate in our current FileSet implementations. I do want to separate the notion of deletions from the File API long term, but this is a pretty low friction way to get this functionality up and running.

Reply:
Can you define a separate delete method and comment it out with a TODO